### PR TITLE
Update README with basic usage example

### PR DESCRIPTION
**Pull request by karolinakuzniewicz**

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3985 Update README for new sdk](https://app.clickup.com/t/2549495/SALEOR-3985)

File: README.md [hunk 1]
```diff
@@ -6,7 +6,136 @@ This package contains methods providing Saleor business logic for a storefront a
...
+
+  if (authenticated && user) {
+    return <span>Signed in as {user.firstName}</span>;
+  } else {
+    return <button onClick={handleSignIn}>Sign in</button>;
+  }
+};
+```
+
+### Custom implementation
```
Comment (krzysztofwolski) [inline README.md:71]:
```suggestion
### Using with NodeJS and other frameworks
```

WDYT? "Custom implementation" sounds to me more like SDK modification, not using it

File: README.md [hunk 2]
```diff
@@ -6,7 +6,136 @@ This package contains methods providing Saleor business logic for a storefront a
...
+
+const client = createSaleorClient({
+  apiUrl: "http://localhost:8000/graphql/,
+  channel: "",
+});
+
+const { auth, config, _internal } = client;
+```
+
+Finally, api methods might be used:
```
Comment (krzysztofwolski) [inline README.md:90]:
```suggestion
Finally, API methods can be used:
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]
Besides comments, lookin' good! Clear examples.
I wish I had it yesterday :D

File: README.md [hunk 3]
```diff
@@ -6,7 +6,136 @@ This package contains methods providing Saleor business logic for a storefront a
...
+
+## Features
+
+We provide an API with methods and fields, performing one, scoped type of work. You may access them straight from `createSaleorClient()` or use React hooks, depending on [which setup do you select](#setup).
+
+| API object | React hook                    | Description                                                                     |
+| :--------- | :---------------------------- | :------------------------------------------------------------------------------ |
+| `auth`     | `useAuth()`, `useAuthState()` | Handles user authentication and stores data about the currently signed in user. |
+| `user`     | `useUser()`                   | Handles user account methods.                                                   |
+| `checkout` | `useCheckout()`               | Handles the whole checkout process.                                             |
```
Comment (MarekChoinski) [inline README.md:118]: We don't have `checkout` yet. It will be `useCheckoutState()` also.

File: README.md (hunk 1 above)
Comment (karolinakuzniewicz) [reply README.md]: sure, changed

File: README.md (hunk 2 above)
Comment (karolinakuzniewicz) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (karolinakuzniewicz) [reply README.md]: That's true but with first stable release we will have both auth and checkout.

**Review by krzysztofwolski — APPROVED** [review]

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -6,7 +6,136 @@ This package contains methods providing Saleor business logic for a storefront a
...
+
+## Features
+
+We provide an API with methods and fields, performing one, scoped type of work. You may access them straight from `createSaleorClient()` or use React hooks, depending on [which setup do you select](#setup).
+
+| API object | React hook                            | Description                                                                     |
+| :--------- | :------------------------------------ | :------------------------------------------------------------------------------ |
+| `auth`     | `useAuth()`, `useAuthState()`         | Handles user authentication and stores data about the currently signed in user. |
+| `user`     | `useUser()`                           | Handles user account methods.                                                   |
+| `checkout` | `useCheckout()`, `useCheckoutState()` | Handles the whole checkout process.                                             |
```
Comment (jwm0) [inline README.md:118]: This doesn't exist yet

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -6,7 +6,136 @@ This package contains methods providing Saleor business logic for a storefront a
...
+```
+
+Use `SaleorProvider` with passed Saleor's client created by `createSaleorClient` in a prop. Then use React hooks in any component passed as child to `SaleorProvider`.
+
+```tsx
+import { SaleorProvider, createSaleorClient, useAuth, useAuthState } from "@saleor/sdk";
+
+const client = createSaleorClient({
+  apiUrl: "http://localhost:8000/graphql/,
+  channel: "",
```
[the remark this answers is not in the document]
Comment (karolinakuzniewicz) [reply README.md]: done

File: README.md [hunk 6]
```diff
@@ -6,7 +6,136 @@ This package contains methods providing Saleor business logic for a storefront a
...
+       * User signed in successfully.
+       **/
+    }
+  };
+
+  if (authenticated && user) {
+    return <span>Signed in as {user.firstName}</span>;
+  } else {
+    return <button onClick={handleSignIn}>Sign in</button>;
+  }
```
[the remark this answers is not in the document]
Comment (karolinakuzniewicz) [reply README.md]: done

File: README.md [hunk 7]
```diff
@@ -6,7 +6,136 @@ This package contains methods providing Saleor business logic for a storefront a
...
+);
+
+const App = () => {
+  const { login } = useAuth();
+  const { authenticated, user } = useAuthState();
+
+  const handleSignIn = async () => {
+    const { data } = await login({email: "admin@example.com", password: "admin"});
+
+    if (data.tokenCreate.errors.length) {
```
[the remark this answers is not in the document]
Comment (karolinakuzniewicz) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (karolinakuzniewicz) [reply README.md:126]: We have pending first checkout pull request, checkout will be available with first official release and lets leave it in readme, not to update it again.

Comment (jwm0) [reply README.md:126]: Remove this for now, apart from it all looks good 🆗

**Review by jwm0 — APPROVED** [review]

**Pull request merged**